gcc/config/riscv/predicates.md

```diff
@@ -161,11 +161,27 @@
 	      && riscv_split_symbol_type (symbol_type)
 	      && symbol_type != SYMBOL_PCREL;
 
+    case POST_MODIFY:
+      if (TARGET_XCVMEM)
+	return false;
+
+    case PLUS:
+      if (TARGET_XCVMEM)
+	return false;
+
     default:
       return true;
     }
 })
 
+(define_predicate "mem_post_inc"
+  (and (match_code "mem")
+       (match_test "GET_CODE (XEXP (op,0)) == POST_MODIFY")))
+
+(define_predicate "mem_plus_reg"
+  (and (match_code "mem")
+       (match_test "GET_CODE (XEXP (op,0)) == PLUS")))
```

Comment:
This is not restrictive enough as it will match `plus (reg) (imm)`.

Reply:
This should be fixed in PR #62